File: src/main/java/com/github/sorusclient/client/hud/impl/armor/IArmorRenderer.java
package com.github.sorusclient.client.hud.impl.armor;

import com.github.sorusclient.client.adapter.IItem;

public interface IArmorRenderer {
    void render(IItem item, double x, double y, double scale);
    void renderArmorPlateBackground(double x, double y, double scale);
    void renderArmorPlate(double x, double y, double scale, ArmorRenderType armorRenderType);

    enum ArmorRenderType {
        FULL,
        HALF,
        EMPTY
    }

}

File: src/main/kotlin/com/github/sorusclient/client/hud/impl/armor/IArmorRenderer.kt
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.hud.impl.armor

import com.github.sorusclient.client.adapter.IItem

interface IArmorRenderer {
    fun render(item: IItem, x: Double, y: Double, scale: Double)
    fun renderArmorPlateBackground(x: Double, y: Double, scale: Double)
    fun renderArmorPlate(x: Double, y: Double, scale: Double, armorRenderType: ArmorRenderType?)
    enum class ArmorRenderType {
        FULL, HALF, EMPTY
    }
}